### DISCUSSION TITLE
Automatic refetch after mutation

### DISCUSSION THREAD

**Discussion by edw19** [urql-graphql/urql#1858] [Q&A] [answered]
hi, I started with urql a few days ago, I have a question about a behavior that you notice, im  configuring urql with cache @urql/exchange-graphcache

I have this interface
![image](https://user-images.githubusercontent.com/47750859/129485649-ff823176-3241-4388-960e-dcd5aa275430.png)

and my config cache
![image](https://user-images.githubusercontent.com/47750859/129485711-9ad4585d-a037-491a-a236-27d2813cef18.png)

the behavior that you notice is that an extra call is made when I run my mutation
![image](https://user-images.githubusercontent.com/47750859/129485822-966efc63-dd57-4c85-af81-41b00b4113bb.png)

and immediately my query is executed
![image](https://user-images.githubusercontent.com/47750859/129485863-6deca4ef-dc9f-447c-b100-80203f6b3e3f.png)

how can i disable that behavior  ? thanks

**Comment by JoviDeCroock** [comment]
That probably means that you are dealing with partial data, it's hard to see why you're refetching but my gut sais that `getProducts` has a more elaborate data requirement than `createProduct` is giving you

**Reply by edw19** [reply]
I think this behavior is from urql because my getProducts query returns a [Products] in graphql and my createProduct mutation returns Product so if I change the return of my createProduct mutation to a String this behavior disappears but I can't use the cache to add the new element

**Reply by JoviDeCroock** [reply]
That's not how graphCache works though, it only refetches if you have partial data 😅 maybe you're using doc-cache in that case it refetches the matching typename

**Reply by edw19** [reply]
maybe it is because i'm storing the data in memory ?

![image](https://user-images.githubusercontent.com/47750859/129487254-a3d354ba-c9c7-4184-8569-a576a19afb6c.png)

**Reply by JoviDeCroock** [reply]
I don't think your problem is situated there, could you try using `updateQuery` instead of `cache.link` 😅 not entirely sure if this one is a good case for `cache.link`

**Reply by edw19** [reply]
dont working 😅😅😅😅😅😅 
it's misspelled ?
![image](https://user-images.githubusercontent.com/47750859/129487785-bf21347d-27b4-4989-beee-6940ad09b446.png)

![image](https://user-images.githubusercontent.com/47750859/129487817-be13dfab-c14e-4b4e-9f96-60c8dfc70928.png)

**Reply by JoviDeCroock** [reply]
mind trying with `__typename`

**Reply by edw19** [reply]
where ?

**Reply by JoviDeCroock** [reply]
Example: https://codesandbox.io/s/peaceful-nightingale-34pfn?file=/src/index.js maybe this helps 😅

**Comment by kitten** [comment, marked as answer]
Can you post some code please or create a reproduction? I can see the thread above, but with just a description of "symptoms" it's a lot harder to understand what's going on.

Maybe post your Client config (including the Graphcache config) and the query and mutation operations here ✌️

(Edit: just clarifying, the full ones for the above; so the screenshots you have are just partial snippets of the above)

**Reply by edw19** [reply]
upload code to github https://github.com/edw19/next-urql-example
thanks for the reply

**Reply by JoviDeCroock** [reply]
You're using the default cache in client-side hydration and the normalized one during ssr

PR to highlight the issue: https://github.com/edw19/next-urql-example/pull/1/files :)

**Reply by edw19** [reply]
thanks for the support that was the error, I will continue to delve into urql 😁😁😁😁😁

**Reply by JoviDeCroock** [reply]
It's a bit of a confusing thing in the Next.JS API's 😅 good luck!

**Reply by MrFarhan** [reply]
Facing the same issue but can not find any proper answer that how to disable this behavior of urql.

The mutation is calling the query, I know I have old / previous / not updated data but I do not need the latest data.

I just want to use the mutation to update the data, whenever I will need updated data so I will run the query but why the mutation is calling the query?
